server/src/main/java/org/elasticsearch/features/FeatureEra.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.features;
+
+import java.util.Arrays;
+
+/**
+ * Represents the major version era a feature was introduced
```

Comment:
Could we omit "eras" from this initial PR, and discuss separately how feature tracking is cleaned up over time? I have some concerns over the complexity and how this would work in serverless, but I don't want to slow this PR from being merged.